src/check.rs

```diff
@@ -94,42 +195,183 @@ fn check_http_url(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
         }
     }
 
-    let resp = ureq::head(url.as_str()).call();
-    if resp.synthetic() {
-        Err(CheckError::Http(Box::new(HttpError::Fetch(
-            url.clone(),
-            resp.into_synthetic_error().unwrap(),
-        ))))
-    } else if resp.ok() {
-        Ok(())
+    // only if the URL contains a fragment we need to fetch the body via
+    // GET, otherwise we can use HEAD
+    if url.fragment().is_none() {
+        let resp = ureq::head(url.as_str()).call();
+
+        handle_response(url, resp).map(|_: ureq::Response| ())
     } else {
-        Err(CheckError::Http(Box::new(HttpError::UnexpectedStatus(
-            url.clone(),
-            resp,
-        ))))
+        // the URL might contain a fragment, in that case we need to check if
+        // the fragment exists, this issues a GET request
```

Comment:
```suggestion
        // The URL might contain a fragment. In that case we need a full GET request to check if the fragment exists.
```
(not sure how to easily do line wrapping in github suggestions)

Reply:
It looks like you resolved this without making the change - did you forget to push commits maybe?